libcfnet/stat_cache.c

```diff
@@ -147,9 +148,17 @@ int cf_remote_stat(AgentConnection *conn, bool encrypt, const char *file,
                  (intmax_t) tloc, file);
         int cipherlen = EncryptString(out, sizeof(out), in, strlen(in) + 1,
                                       conn->encryption_type, conn->session_key);
+
+        tosend = cipherlen + CF_PROTO_OFFSET;
+
+        if(tosend > sizeof(sendbuffer))
+        {
+            ProgrammingError("cf_remote_stat: tosend (%d) > sendbuffer (%ld)",
+                             tosend, sizeof(sendbuffer));
```

Comment:
Yes, I agree. Even if there might be better ways to handle this, it's better than it was before.